src/veh_interact.cpp

```diff
@@ -371,6 +378,15 @@ void veh_interact::cache_tool_availability()
                 crafting_inv.has_components( "wheel_bicycle", 1 ) ||
                 crafting_inv.has_components( "wheel_motorbike", 1 ) ||
                 crafting_inv.has_components( "wheel_small", 1 );
+
+    int qual_jack = ceil( double( veh->total_mass() * 1000 ) / TOOL_LIFT_FACTOR );
+    has_jack = g->u.has_quality( "JACK", qual_jack ) ||
+               map_selector( g->u.pos(), PICKUP_RANGE ).has_quality( "JACK", qual_jack ) ||
+               vehicle_selector( g->u.pos(), 1, *veh ).has_quality( "JACK", qual_jack );
```

Comment:
Deliberately so. Vehicle mounted cranes are somewhat fragile and the heavier variants need to be mounted as external protrusions. Forcing the player to maneuver to the target prevents exploits like hiding the crane in the middle of the vehicle.